ngraph/core/src/op/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,175 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "itt.hpp"
+
+#include "ngraph/op/ctc_greedy_decoder_seq_len.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v6::CTCGreedyDecoderSeqLen, "CTCGreedyDecoderSeqLen", 6);
+
+op::v6::CTCGreedyDecoderSeqLen::CTCGreedyDecoderSeqLen(const Output<Node>& input,
+                                                       const Output<Node>& seq_len,
+                                                       const bool merge_repeated,
+                                                       const element::Type& classes_index_type,
+                                                       const element::Type& sequence_length_type)
+    : Op({input, seq_len})
+    , m_merge_repeated(merge_repeated)
+    , m_classes_index_type(classes_index_type)
+    , m_sequence_length_type(sequence_length_type)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v6::CTCGreedyDecoderSeqLen::CTCGreedyDecoderSeqLen(const Output<Node>& input,
+                                                       const Output<Node>& seq_len,
+                                                       const Output<Node>& blank_index,
+                                                       const bool merge_repeated,
+                                                       const element::Type& classes_index_type,
+                                                       const element::Type& sequence_length_type)
+    : Op({input, seq_len, blank_index})
+    , m_merge_repeated(merge_repeated)
+    , m_classes_index_type(classes_index_type)
+    , m_sequence_length_type(sequence_length_type)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v6::CTCGreedyDecoderSeqLen::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v6_CTCGreedyDecoderSeqLen_validate_and_infer_types);
+    const auto& logits_pshape = get_input_partial_shape(0);
+    const auto& seq_len_pshape = get_input_partial_shape(1);
+    auto input_et = get_input_element_type(0);
+    const bool logits_is_static_rank = logits_pshape.rank().is_static();
+    const bool seq_len_is_static_rank = seq_len_pshape.rank().is_static();
+
+    // output dynamic rank tensor if all inputs are of dynamic rank
+    if (!logits_is_static_rank && !seq_len_is_static_rank)
+    {
+        set_output_type(0, input_et, PartialShape{Dimension::dynamic(), Dimension::dynamic()});
+    }
```

Comment:
This code is not needed because you always set output types in lines 134 and 135.